### PR TITLE
feat: request mode in raw SQL

### PR DESCRIPTION
**Pull request by aceforeverd**
##  request mode in RAW SQL
```sql
SELECT ....
CONFIG ( execute_mode = 'request', values = (expr1, expr2 , ...) )

SELECT ....
CONFIG ( execute_mode = 'request', values = [(expr1, expr2 , ...), (....)] )
```

## call procedure 

```sql
call dp (expr1, expr2, ...)
```

## TODOs
- docs
- support `request` value in system variable `execute_mode`
- take care const query

### CODE REVIEW

**Comment by aceforeverd** [comment]
TODO:
- auto infer from CONFIG values data type

File: hybridse/src/vm/engine.cc [hunk 1]
```diff
@@ -383,12 +383,64 @@ bool RunSession::SetCompileInfo(const std::shared_ptr<CompileInfo>& compile_info
...
+    ::hybridse::codec::Row row = in_row;
+    std::vector<::hybridse::codec::Row> sql_request_rows;
+    // extract request values in SQL itself,
+    auto s = ExtractRequestRowsInSQL(&sql_request_rows);
+    if (!s.ok()) {
+        LOG(WARNING) << "extract request error: " << s;
+        return -1;
+    }
+    if (!sql_request_rows.empty()) {
+        row = sql_request_rows.at(0);
```
Comment (aceforeverd) [inline hybridse/src/vm/engine.cc:463]: add compile time check

**Comment by aceforeverd** [comment]
execute_mode rules
-----

## Values
`execute_mode` in **SQL CONFIG** clause can be
- online
- request
- offline
- *batchrequest (hidden, may removed in later release when 'request' feature complete. Users are generally aware about the three options above )*

`values` option in **SQL CONFIG** clause can be
- StructConstructorWithParens ( e.g `( expr1, expr2, ...)` )
- Array [ StructConstructorWithParens ] ( e.g `[ (expr1, expr2, ...), ( ...), ... ]` )

`execute_mode` in **system variable** can be
- online
- offline
- request ( TODO later, won't support in this PR)


## Rules
- determine `execute_mode`
  - take `execute_mode` in CONFIG clause IF EXISTS
  - OTHERWISE take `execute_mode` from system variable `execute_mode` (pass in from sql_cluster_router client)
- `execute_mode` to `engine_mode` in SQL compiler
  - online -> batch
  - offline -> offline
  - request -> 
     - `values` defines single row: request 
     - `values` defines multiple row: batchrequest
  - batchrequest -> batchrequest
- Special cases
  - `engine_mode = 'request' | 'batchrequest'` and request table not empty
    - **COMPILE ERROR**: when  CONFIG `values` does not match expected schema of request table
    - **COMPILE ERROR**: when CONFIG `values` is empty array
  - `engine_mode = 'request' | 'batchrequest'` and request table is empty (a const query without table)
    - CONFIG `values` can be empty struct or empty array. 
    - This PR won't take care of this case, TODO later
- CONFIG `values` ignored if `engine_mode` is `batch` or `offline`

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**